metadata-io/src/main/java/com/linkedin/metadata/client/JavaEntityClient.java

```diff
@@ -784,17 +784,17 @@ public List<String> batchIngestProposals(
     // Preserve ordering
     return batch.getItems().stream()
         .map(
-            requestItem -> {
-              if (resultMap.containsKey(requestItem)) {
-                List<IngestResult> results = resultMap.get(requestItem);
-                return results.stream()
-                    .filter(r -> r.getUrn() != null)
+            requestItem ->
```

Comment:
A bit confused on this, we're swapping over to processing the entry set instead of doing a contains key which would be more efficient? If it's due to the equals impl not matching would it make more sense to override the equals method? Or should the map key be changed?